pytorch_lightning/trainer/data_loading.py

```diff
@@ -293,7 +293,13 @@ def _get_dataloader_init_kwargs(
             dl_kwargs["batch_sampler"] = None
             dl_kwargs["sampler"] = None
 
-        if _fault_tolerant_training():
+        dl_kwargs = TrainerDataLoadingMixin._prepare_fault_tolerance(dl_kwargs)
+
+        return dl_kwargs
+
+    @staticmethod
+    def _prepare_fault_tolerance(dl_kwargs: Dict) -> Dict:
+        if _fault_tolerant_training_mode().is_automatic:
```

Comment:
The name is misleading if the body only gets executed for some modes. I propose:

```py
if _fault_tolerant_training_mode().is_automatic:
    dl_kwargs["dataset"] = _capture_dataset(dl_kwargs["dataset"])


def _capture_dataset(...):
    ...
```

Reply:
Done !